libqtile/widget/pulse_volume.py

```diff
@@ -221,17 +222,20 @@ def update(self):
         manually re-schedule update
         """
         vol = self.get_volume()
-        if vol != self.volume:
+        if vol != self.volume or self.mute is not self._old_mute:
             self.volume = vol
+            self._old_mute = self.mute
             # Update the underlying canvas size before actually attempting
             # to figure out how big it is and draw it.
             self._update_drawer()
             self.bar.draw()
 
     def get_volume(self):
+        self.mute = False
         if self.default_sink:
             if self.default_sink["muted"]:
-                return -1
+                self.mute = True
+                return 0
```

Comment:
We can remove the `return 0` line. It should be fine to return the actual volume level as we've set the `mute` flag.